services/payments/controllers.go

```diff
@@ -126,21 +148,21 @@ func SubmitHandler(service *Service) handlers.AppHandler {
 
 		var (
 			logger = logging.Logger(ctx, "SubmitHandler")
-			req    = &Transaction{}
+			authenticatedState    = &AuthenticatedPaymentState{}
```

Comment:
again, we're reading directly into the `AuthenticatedPaymentState`. shouldn't this be `SubmitRequest`?